website/content/docs/runtime/interpolation.mdx

```diff
@@ -43,6 +48,17 @@ task "docs" {
     value     = "linux"
   }
 
+  template {
+    destination = "template.txt"
+    data = <<EOT
+{{- /*
+      Environment variables are available to templates via the env function,
```

Comment:
Hm, I didn't realize we referred to Nomad's builtin variables as "environment variables" in this doc. Seems really confusing with what most people think of as "environment variables." We now have 2 definitions for that phrase in our docs:

1. The classic `USUALLY_UPPER_CASE` environment variables supported by every operating system I know of
2. Nomad's "environment" variables which are allocation-specific _and include env vars from definition `#1`_

Not a blocker for your PR but it would be nice to come up with a distinct term for `#2`. Hopefully "variable" is unambiguous enough, but if not perhaps "Job variables" or "Jobspec variables"?

Reply:
Yeah, there's definitely something to be said around the node variables (which are only available in the alloc runner environment) and the alloc variables which are the NOMAD_ variables that are passed to and visible in the workload.

Created https://github.com/hashicorp/nomad/issues/11201 to track this.